Source/Core/Common/GL/GLInterface/BGL.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/GL/GLInterface/BGL.h"
+
+#include <GLView.h>
+#include <Size.h>
+#include <Window.h>
+
+#include "Common/Assert.h"
+
+BGLView* GLContextBGL::m_current = nullptr;
+
+GLContextBGL::~GLContextBGL()
+{
+  if (!m_window)
+    delete m_gl;
+}
+
+bool GLContextBGL::Initialize(const WindowSystemInfo& wsi, bool stereo, bool core)
+{
+  m_window = static_cast<BWindow*>(wsi.render_window);
+
+  m_gl = new BGLView(m_window ? m_window->Bounds() : BRect(), "GLContextBGL", B_FOLLOW_ALL_SIDES, 0,
+                     BGL_RGB | BGL_DOUBLE | BGL_ALPHA);
+  if (m_window)
+    m_window->AddChild(m_gl);
+
+  m_opengl_mode = Mode::OpenGL;
+
+  m_gl->LockLooper();
+  BRect size = m_gl->Frame();
+  m_gl->UnlockLooper();
+
+  m_backbuffer_width = size.IntegerWidth();
+  m_backbuffer_height = size.IntegerHeight();
+
+  MakeCurrent();
+
+  return true;
+}
+
+bool GLContextBGL::IsHeadless() const
+{
+  return !m_window;
```

Comment:
In this particular case, it might be clearer to write the check like this
```suggestion
  return m_window == nullptr;
```
The combination of the ! and the "is ..." function name makes it look like you're inverting a boolean.